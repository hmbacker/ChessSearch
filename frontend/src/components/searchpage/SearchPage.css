.searchpage {
  text-align: center;
}

.searchpage-header {
  min-height: 20vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.searchpage-section {
  align-items: center;
  justify-content: center;
}

.graph-container {
  max-height: 250px;

}

.searchpage-footer {
  align-items: center;
  justify-content: center;
  color: white;
}

.image-chess {
  max-height: 30vh;
  size: calc(10px + 2vmin);
}

footer a:link {
  color: skyblue;
}

footer a:visited {
  color: rgb(113, 177, 202);
}