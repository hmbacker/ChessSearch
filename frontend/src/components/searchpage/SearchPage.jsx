import React from "react";
import "./SearchPage.css";
import chess_pieces from "../../media/chess_pieces.svg";
import SearchBox from "../searchbox/SearchBox";
import ResultBox from "../resultbox/ResultBox";
import Graph from '../advancedview/Graph'

function SearchPage() {
  return (
    <div className="searchpage">
      <header className="searchpage-header">
        <h1>Chess Search</h1>
      </header>
      <section className="searchpage-section">
        <img
          src={chess_pieces}
          className="image-chess"
          alt="white chess knight"
        />
        <SearchBox />
        <Graph />
        <ResultBox />
      </section>
      <footer className="searchpage-footer">
        <div>
          Loading icon made by{" "}
          <a
            href="https://www.flaticon.com/authors/roundicons"
            title="Roundicons"
            target="_blank"
            rel="noopener noreferrer"
          >
            Roundicons
          </a>{" "}
          from{" "}
          <a
            href="https://www.flaticon.com/"
            title="Flaticon"
            target="_blank"
            rel="noopener noreferrer"
          >
            www.flaticon.com
          </a>
        </div>
      </footer>
    </div>
  );
}

export default SearchPage;
