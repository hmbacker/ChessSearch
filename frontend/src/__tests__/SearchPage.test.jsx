import React from "react";
import SearchPage from "../components/searchpage/SearchPage";
//import store from "../store";
import { shallow } from "enzyme";

test("renders correctly", () => {
  const wrapper = shallow(<SearchPage />);
  expect(wrapper.debug()).toMatchSnapshot();
});
