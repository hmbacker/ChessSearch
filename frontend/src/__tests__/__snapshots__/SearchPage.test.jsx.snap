// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
"<div className=\\"searchpage\\">
  <header className=\\"searchpage-header\\">
    <h1>
      Chess Search
    </h1>
  </header>
  <section className=\\"searchpage-section\\">
    <img src={{...}} className=\\"image-chess\\" alt=\\"white chess knight\\" />
    <Connect(SearchBox) />
    <Connect(Graph) />
    <Connect(ResultBox) />
  </section>
  <footer className=\\"searchpage-footer\\">
    <div>
      Loading icon made by
       
      <a href=\\"https://www.flaticon.com/authors/roundicons\\" title=\\"Roundicons\\" target=\\"_blank\\" rel=\\"noopener noreferrer\\">
        Roundicons
      </a>
       
      from
       
      <a href=\\"https://www.flaticon.com/\\" title=\\"Flaticon\\" target=\\"_blank\\" rel=\\"noopener noreferrer\\">
        www.flaticon.com
      </a>
    </div>
  </footer>
</div>"
`;
